core/src/services/sqlite/backend.rs

```diff
@@ -0,0 +1,294 @@
+        let value_field = match self.value_field.clone() {
+            Some(v) => v,
+            None => "value".to_string(),
+        };
+        let root = normalize_root(
+            self.root
+                .clone()
+                .unwrap_or_else(|| "/".to_string())
+                .as_str(),
+        );
+        Ok(SqliteBackend::new(Adapter {
+            connection_string,
+            table,
+            key_field,
+            value_field,
+        })
+        .with_root(&root))
+    }
+}
+
+pub type SqliteBackend = kv::Backend<Adapter>;
+
+#[derive(Clone)]
+pub struct Adapter {
+    connection_string: String,
+
+    table: String,
+    key_field: String,
+    value_field: String,
+}
+
+impl Debug for Adapter {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        let mut ds = f.debug_struct("SqliteAdapter");
+        ds.field("connection_string", &self.connection_string);
+        ds.field("table", &self.table);
+        ds.field("key_field", &self.key_field);
+        ds.field("value_field", &self.value_field);
+        ds.finish()
+    }
+}
+
+#[async_trait]
+impl kv::Adapter for Adapter {
+    fn metadata(&self) -> kv::Metadata {
+        kv::Metadata::new(
+            Scheme::Sqlite,
+            &self.table,
+            Capability {
+                read: true,
+                write: true,
+                create_dir: true,
+                delete: true,
+                ..Default::default()
+            },
+        )
+    }
+
+    async fn get(&self, path: &str) -> Result<Option<Vec<u8>>> {
+        let connection_string = self.connection_string.clone();
+        let value_field = self.value_field.clone();
+        let table = self.table.clone();
+        let key_field = self.key_field.clone();
+        let cloned_path = path.to_string();
+
+        task::spawn_blocking(move || {
+            let query = format!(
+                "SELECT {} FROM {} WHERE `{}` = $1 LIMIT 1",
+                value_field, table, key_field
+            );
+            let conn = Connection::open(connection_string).map_err(Error::from)?;
+            let mut statement = conn.prepare(&query).map_err(Error::from)?;
+            let result = statement.query_row([cloned_path.as_str()], |row| row.get(0));
+            match result {
+                Ok(v) => Ok(Some(v)),
+                Err(rusqlite::Error::QueryReturnedNoRows) => Ok(None),
+                Err(err) => Err(Error::from(err)),
+            }
+        })
+        .await
+        .map_err(Error::from)
+        .and_then(|inner_result| inner_result)
+    }
+
+    async fn set(&self, path: &str, value: &[u8]) -> Result<()> {
+        let connection_string = self.connection_string.clone();
+        let table = self.table.clone();
+        let key_field = self.key_field.clone();
+        let value_field = self.value_field.clone();
+        let cloned_path = path.to_string();
+        let cloned_value = value.to_vec();
+
+        task::spawn_blocking(move || {
```

Comment:
I think it's better to implement blocking_get and call it here so that sqlite can have blocking support too.

This can be implemented in next PR if you want.